src/core_plugins/metrics/server/lib/get_fields.js

```diff
@@ -1,40 +1,9 @@
-import _ from 'lodash';
+import { sortBy, uniq } from 'lodash';
 
-export function getParams(req) {
+export async function getFields(req) {
+  const { indexPatternsService } = req.pre;
   const index = req.query.index || '*';
-  return {
-    index,
-    fields: ['*'],
-    ignoreUnavailable: false,
-    allowNoIndices: false,
-    includeDefaults: true
-  };
+  const resp = await indexPatternsService.getFieldsForWildcard({ pattern: index });
+  const fields = resp.filter(field => field.aggregatable);
+  return sortBy(uniq(fields, field => field.name), 'name');
```

Comment:
There isn't any harm in sorting the fields, but just so you know that's how they are returned by the `getFieldsForWildcard()` API https://github.com/elastic/kibana/blob/master/src/server/index_patterns/service/lib/field_capabilities/__tests__/field_capabilities.js#L67-L77